### DISCUSSION TITLE
Field not in the DB

### DISCUSSION THREAD

**Discussion by dfioravanti** [jowilf/starlette-admin#200] [Q&A] [answered]
Hey everyone, I was wondering if this project supports defining columns at runtime as `flask-admin` does. What I mean is that supposed we have a model like

```python
class Table(Base):
   id: int
   name: str
```

and I would like to show a column in the `List` page that is called `status` that would depend on the `name`. For example `status == True` if `active` is in `name`. With `flask-admin` all I need to do is define a formatter and then add that formatter to the `column_formatters` dictionary. Is it possible to do something similar in this project?

**Comment by jowilf** [comment, marked as answer]
you can create a custom field and override the [parse_obj](https://jowilf.github.io/starlette-admin/api/fields/#starlette_admin.fields.BaseField.parse_obj) function.

Example:
```py
class MyCustomField(BooleanField):

    async def parse_obj(self, request: Request, obj: Any) -> Any:
        return 'active' in obj.name
```
then you can define your field list like this: 
```py
fields = ['id', 'name', MyCustomField('status', exclude_from_create=True, exclude_from_edit=True)]
```

Full example -> https://github.com/jowilf/starlette-admin-demo/blob/ad82fde9acf7dbd5a2cf58d51ec2e3db61451ba6/app/sqla/fields.py#L24

For more information -> https://jowilf.github.io/starlette-admin/advanced/custom-field/

**Reply by dfioravanti** [reply]
Oh amazing, thank that helps a lot
